### PR TITLE
LPS-169695 [Raylife AP] Create Filter Request

### PR DESCRIPTION
**Pull request by eduardo-diniz**
[LPS-169695](https://issues.liferay.com/browse/LPS-169695)

### CODE REVIEW

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/utils/dateFormatter.js [hunk 1]
```diff
@@ -39,16 +44,92 @@ export const december = '12';
...
+	)
+).split('-');
+
+export const sixMonthsAgoDateLastYear = convertDateToString(
+	new Date(
+		new Date(
+			new Date(new Date().setFullYear(lastYear)).setMonth(sixMonthsAgo)
+		).setDate(fistDayOfMonth)
+	)
+).split('-');
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/utils/dateFormatter.js:117]: Try to do it in a generic way, like this:

```suggestion

const generateDateThroughPeriod = (year, period) =>{
     const generatedDate = convertDateToString(
	new Date(
		new Date(
			new Date(new Date().setFullYear(year)).setMonth(period)
		).setDate(fistDayOfMonth)
	)
).split('-');

   return generatedDate;
}

export const sixMonthsAgoDateLastYear = generateDateThroughPeriod(lastYear,sixMonthsAgo)
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/claims-total-settlements-paid/index.js [hunk 2]
```diff
@@ -15,18 +15,553 @@
...
+				if (month in claimElement) {
+					claimElement[month] += claim?.claimAmount;
+				}
+			});
+		});
+
+		return claimsArray;
+	}
+
+	const getMonthsSettlementPaid = (response, sixMonthsAgoArray) => {
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/claims-total-settlements-paid/index.js:104]:
```suggestion
	const getMonthsSettlementPaid = (response, arrayOfMonths) => {
```

**Review by joanagrego — CHANGES_REQUESTED** [review]
Nice job! I added some suggestions :)

**Review by silasbsi — APPROVED** [review]
LGTM :heavy_check_mark: You need to do the suggestions made by @joanagrego. Let me know when you're done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/services/Claim.js [hunk 3]
```diff
@@ -19,3 +19,35 @@ const DeliveryAPI = 'o/c/raylifeclaims';
...
+	currentMonth,
+	currentDay,
+	periodYear,
+	periodMonth,
+	periodDay
+) {
+	return axios.get(
+		`${DeliveryAPI}/?fields=claimStatus,claimCreateDate,claimAmount&pageSize=200&filter=claimCreateDate ge ${currentYear}-${currentMonth}-${currentDay} and claimCreateDate le ${periodYear}-${periodMonth}-${periodDay}`
+	);
+}
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/services/Claim.js:47]: Done
Thx!

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/utils/dateFormatter.js (hunk 1 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/utils/dateFormatter.js:117]: Done.

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Review by joanagrego — APPROVED** [review]
Approved! Thank you @eduardo-diniz :white_check_mark:

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by joanagrego** [comment]
ci:test:relevant

**Comment by joanagrego** [comment]
ci:test:relevant

**Comment by silasbsi** [comment]
ci:forward

**Pull request closed without merging**